### ISSUE TITLE
Compatibility issue between prosemirror-tables and new Prosemirror version that include TypeScript definitions

### ISSUE THREAD

**Issue by lpellegr** [ProseMirror/prosemirror-tables#171]
For instance, when upgrading from prosemirror-schema version 1.16.1 to version 1.17.0, type definitions are no longer the same. This leads to many errors.

```
[tsc] node_modules/prosemirror-tables/index.d.ts(62,28): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(62,56): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(66,10): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(67,8): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(70,12): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(71,10): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(72,16): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(73,14): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(75,17): error TS2315: Type 'SelectionRange' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(77,12): error TS2315: Type 'Node' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(78,14): error TS2315: Type 'Slice' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(79,15): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(79,40): error TS2315: Type 'Slice' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(80,19): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(80,41): error TS2315: Type 'Node' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(81,25): error TS2315: Type 'Node' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(84,13): error TS2315: Type 'Selection' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(89,17): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(90,16): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(93,17): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(94,16): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(97,10): error TS2315: Type 'Node' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(102,10): error TS2315: Type 'Node' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(139,10): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(140,19): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(145,12): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(145,44): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(148,10): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(149,19): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(153,10): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(154,19): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(158,10): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(159,19): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(169,12): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(169,44): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(174,12): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(174,44): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(177,10): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(178,19): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(188,49): error TS2315: Type 'NodeType' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(189,12): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(189,44): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(192,10): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(193,19): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(197,10): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(198,19): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(202,10): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(207,10): error TS2315: Type 'Node' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(212,10): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(213,19): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(217,10): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(218,19): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(222,16): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(225,4): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(228,10): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(229,19): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(233,10): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(234,19): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(238,10): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(239,19): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(243,16): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(246,4): error TS2315: Type 'Transaction' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(251,10): error TS2315: Type 'NodeView' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(269,8): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(270,4): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(283,10): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(284,4): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(287,8): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(288,4): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(291,7): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(292,7): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(302,8): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(305,11): error TS2315: Type 'ResolvedPos' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(308,10): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(309,14): error TS2315: Type 'EditorState' is not generic.
[tsc] node_modules/prosemirror-tables/index.d.ts(310,11): error TS2315: Type 'Transaction' is not generic.
```

**Comment by lpellegr** [comment]
Related to https://github.com/ProseMirror/prosemirror-tables/pull/170

**Comment by lpellegr** [comment]
You might be right but I remember a discussion that prosemirror-tables is maintained independently, mainly by guys from Atlassian.

**Comment by MH4GF** [comment]
I asked marijn before and here is the result, which seems to be maintained by atlassian. ref: https://discuss.prosemirror.net/t/why-is-prosemirror-tables-not-maintained/4522

@RatoX hey, you have recently introduced prettier and eslint, do you plan to resolve this issue? If so, I am very happy to hear the news. (If not, I am going to fork and solve the problem.)

**Comment by ocavue** [comment]
Fixed in v1.2.1
